### ISSUE TITLE
Height integration

### ISSUE THREAD

**Issue by nichollsh** [FormingWorlds/JANUS#12]
In some cases the height integration blows up.

This is particularly bad when the atmosphere has a low MMW and a high temperature, so integrating down to pressures ~1 Pa leads to scale heights ~1e9 metres. This effectively represents a state where the atmosphere is unbound.

We know the height integration works fine, because it usually gets reasonable values. I've also tested it for Jupiter-like cases (pure H2, cold) and it gets scale heights ~50 km. It also compares well against RFM retrievals.

Since the cell heights are only needed for plotting (at the moment) we could simply not calculate them.

**Comment by timlichtenberg** [comment]
Alternatively, you just set a simple check for when this happens (and log the PROTEUS step), and then prevent plotting. I say this because I find this very useful, and – indeed – at the moment it is just optics, but it in principle computes the planet total  / transit size.

**Comment by nichollsh** [comment]
Makes sense. It would probably be best if this flag is included in JANUS directly, since that's where the heights are calculated. I'll make this change once the cloud stuff is merged. It will then require a small change to PROTEUS to log when this is caught by JANUS.

**Comment by nichollsh** [comment]
Note that this issue is also found to occur in modelling by Maxime Maurice in [this work](https://doi.org/10.1051/0004-6361/202347749).

> No realistic transit radii for H-rich reduced small planets could be obtained because the assumption of hydrostatic equilibrium breaks down in these conditions of surface pressure, temperature, and gravity.

**Comment by timlichtenberg** [comment]
There should be a warning inserted into the LOGGER when this happens.

**Comment by nichollsh** [comment]
Turns out that JANUS is already printing a warning to the log when this occurs! See this line:
https://github.com/FormingWorlds/JANUS/blob/be69d398e1bb580aa9f85d3a297ad0ec9e4e9e7b/src/janus/utils/height.py#L44

When this happens, the height array is set to dummy values between 0 and 1 km, which I think is fine. PROTEUS should probably record when this happens, but that can be a separate issue.
